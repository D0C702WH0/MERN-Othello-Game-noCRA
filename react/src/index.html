<!DOCTYPE html>
<html lang="fr">
  <head>
    <title>zoOthello</title>
    <base href="/"/>
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="stylesheet" href="style.css"/>
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css"
    integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous" />
    <link href="https://fonts.googleapis.com/css?family=Patua+One&display=swap" rel="stylesheet"/>    
    <link href="https://fonts.googleapis.com/css?family=Roboto&display=swap" rel="stylesheet"/>
  </head>
  <body>
    <div id="root"></div>
  </body>
</html>