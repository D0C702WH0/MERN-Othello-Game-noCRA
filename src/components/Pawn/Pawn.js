import React from 'react'
import PropTypes from 'prop-types'

import './Pawn.scss'

export default function Pawn ({ color }) {
  return (
    <>
      {
        color !== '' &&
        <img alt="duck pawn" data-testid={`test-pawn-${color === 'black' ? 'red' : 'yellow'}`} className={`duckFace ${color === 'black' ? 'red' : 'yellow'}`} src={`./assets/img/duck-${color === 'black' ? 'red' : 'yellow'}.png`}/>
      }
    </>
  )
}

Pawn.propTypes = {
  color: PropTypes.string
}
