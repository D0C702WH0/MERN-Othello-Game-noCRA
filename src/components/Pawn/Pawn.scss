
.duckFace {
  width: 3rem;
  margin: 0 auto;
}

@media only screen and (min-width:0px) and (max-width:767px) {
  .duckFace {
    width : 1.5rem;
  }
}