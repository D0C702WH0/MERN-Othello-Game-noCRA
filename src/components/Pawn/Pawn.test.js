import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Pawn from './Pawn'

let wrapper
beforeEach(() => {
  wrapper = render(
    <Pawn color={'black'}/>
  )
})

describe('Pawn', () => {
  it('Basic rendering', () => {
    const { asFragment } = wrapper
    expect(asFragment()).toMatchSnapshot()
  })

  it('Should have "red" as a className', () => {
    const { getByTestId } = wrapper
    expect(getByTestId('test-pawn')).toHaveClass('red')
  })

  it('Should change "red" className to "yellow" on props change', () => {
    const { rerender, getByTestId } = wrapper
    rerender(<Pawn color={'white'} />)
    expect(getByTestId('test-pawn')).toHaveClass('yellow')
  })
})
