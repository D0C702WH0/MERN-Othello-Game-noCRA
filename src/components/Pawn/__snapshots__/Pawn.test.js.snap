// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pawn Basic rendering 1`] = `
<DocumentFragment>
  <img
    alt="duck pawn"
    class="duckFace red"
    data-testid="test-pawn"
    src="./assets/img/duck-red.png"
  />
</DocumentFragment>
`;
